### DISCUSSION TITLE
[Issue] Copilot not working on VS Code

### DISCUSSION THREAD

**Discussion by s0r3-glitch** [community/community#6707] [Copilot Conversations]
I installed copilot on a second instance of VS code so I could have a portable version of it, however, the extension does not show up in the status bar and none of the keybinds work. I have linked my github to my VS code instance but am unsure what to do at this point. Does anyone know where I can look for the extension logs or what i can do to get it working?

Edit: I am using VS code 1.61.2, with copilot v1.5.3432, on windows_NT x64 10.0.18363

Update: I have been looking around the feedback to see if anyone has a solution to this and it appears that copilot isnt even starting since I dont see the icon in the status bar. The extension is enabled and my github is linked so I am not sure what to do now.

Update 2: I have updated copilot to 1.6.3440 and its still not showing up in the status bar. None of the commands are registering either.

Update 3: Finally got copilot to work, I had to go back and install version 1.1.1909 in order to get it to work. It seems like all versions before that work as well so something in 1.1.1940 is preventing it from working.

**Comment by ZachariBarnes** [comment]
Seeing similar issues here. Installed on my second computer and it shows in the status bar but no suggestions appear and the keybinds do not work.

**Reply by s0r3-glitch** [reply]
My instance doesn't even have it show in the status bar. Hopefully someone comes with a fix for this because I would love to get this working on my laptop.

**Reply by Jobrien0394** [reply]
Yeah, mine shows up in the status bar, and works, however mine does 1 line at a time rather than a whole block of code and i have to hit tab to accept it, it doesn't even display a setting for a next suggestion, previous suggestion, etc. However I do have Tabnine installed as well, so I'm wondering if it's conflicting with that extension.

**Comment by hsenag** [comment]
Hi, sorry this is going wrong for you. I've just double-checked on my Windows machine and it is working for me. VSCode 1.61.2, Copilot v1.6.3440.

Could you have a look if there is any useful information in the Output window? (select GitHub Copilot in the right-hand drop down)

**Reply by s0r3-glitch** [reply]
Just checked and there is nothing in the github copilot output window. I can confirm that the extension is enabled as its showing me the time it takes to start but its just not showing it in the status bar or registering commands.

**Reply by hsenag** [reply]
Could you try uninstalling and reinstalling it, if you didn't already? It's also surprising that 1.1.1909 works as it's very old and things have changed on the server-side since then.

**Reply by s0r3-glitch** [reply]
I have tried reinstalling it multiple times with no success. The only thing that 1.1.1909 does in the sense of it "working" is have copilot show up in the status bar.

**Reply by Akxe** [reply]
I have the same problem...
* Both of my computers stopped working at the same time
* I reinstalled the extension on both PC (multiple times)
* I have no logs in VS code's output for GitHub Copilot
* I have tested via Wireshark, that a request got sent out. (first certificate, then a request/response)

But I got only `Synthesizing 0/10 solutions`

**Comment by ZachariBarnes** [comment]
The copilot output for my case (installed and in status bar but not working) is as follows
`[INFO] [default] [2021-10-25T19:18:43.073Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[ERROR] [default] [2021-10-25T19:18:43.450Z] Request Error: unable to verify the first certificate
[ERROR] [ghostText] [2021-10-25T19:18:43.450Z] Error on ghost text request Error: unable to verify the first certificate
[INFO] [ghostText] [2021-10-25T19:18:43.450Z] Breaking, no choices
`

**Reply by s0r3-glitch** [reply]
are you using latest or an old version?

**Reply by ZachariBarnes** [reply]
Latest I believe. v1.6.3440

**Reply by s0r3-glitch** [reply]
I had a similar error when running v1.1.1940 but i have no idea how to fix that. Id open a new discussion.

**Reply by davecheney** [reply]
Are you behind a corporate firewall or some other device that may be intercepting your SSL sessions?

**Reply by s0r3-glitch** [reply]
> Are you behind a corporate firewall or some other device that may be intercepting your SSL sessions?

in my case I am behind a corporate firewall and was getting self signed cert errors.

**Reply by ZachariBarnes** [reply]
I am indeed behind a firewall. I suspected this was the case.

**Reply by davecheney** [reply]
Sorry folks, you'll have to talk to the folks that run those devices and let them know that their MITM isn't as transparent as they'd like.

**Reply by s0r3-glitch** [reply]
> I am indeed behind a firewall. I suspected this was the case.

Just out of curiosity, did v1.6.3440 fix your issue with copilot not showing up in the status bar?

**Reply by kahunacoder** [reply]
I'm having this issue too but I'm not behind a firewall. VS code 1.61.2, CoPilot v1.6.3440, Mac OS. No icon and all commands fail.

**Reply by davecheney** [reply]
@kahunacoder what do you see in the github copilot output window?

**Reply by kahunacoder** [reply]
I don't see a "github copilot output window"... How do I find this?

**Reply by kahunacoder** [reply]
I deleted `vscodevscode.github-authentication` from my keychain and was able to re-authenticate. First time I authenticated I never saw the `agree to terms` notification. I'm excited to git started!

**Reply by linogarciavallejo** [reply]
> I deleted `vscodevscode.github-authentication` from my keychain and was able to re-authenticate. First time I authenticated I never saw the `agree to terms` notification. I'm excited to git started!

I tried your approach @kahunacoder but didn't work for me. I can't see the github copilot icon in my VSCode :(

**Reply by kahunacoder** [reply]
@linogarciavallejo Before I deleted my auth key I reinstalled the extensions and vscode and restarted vscode as well as my computer. I'm not sure what actually worked but reauth was the last thing I did.

**Reply by rr1000** [reply]
Where do you delete vscodevscode.github-authentication?

**Reply by kahunacoder** [reply]
@rr1000 I'm on a Mac and it was in `Macintosh HD:System:Applications:Utilities:Keychain Access.app`.

**Reply by s0r3-glitch** [reply]
> Sorry folks, you'll have to talk to the folks that run those devices and let them know that their MITM isn't as transparent as they'd like.

Are you able to provide the copilot server URL's or IP addresses so that I can make a request to my organization to allow communication to those servers?

**Comment by RacketyWater7** [comment]
So it happened to me as well, and what I did is I went into the Extensions of Vs Code; disabled the copilot extension, reloaded the Code. And (re-) enabled the extension after reloading. And the copilot started. Hope that helps!

**Comment by emmanuel-galindo** [comment]
same here, behind zscaler proxy and no info in the output console .. 

I am in windows, and I've removed the credentials and triggered the auth process again.. when github sends the token back to vscode, the extension authenticates, the extesion reports as active, but does not show the icon in the bottom and logs nothing else 

other extensiosn work ok behind proxy

**Reply by emmanuel-galindo** [reply]
today I just noticed the icon appeared in the bottom, but with a strike. Logs now also show;
[ERROR] [default] [2021-12-01T12:31:35.810Z] GitHub Copilot could not connect to server. Extension activation failed: "read ECONNRESET"

I tried a couple of changes in vscode proxy settings w/out luck though

**Reply by darshanhande11** [reply]
@emmanuel-galindo Same error popped up when I read the output log of GitHub Copilot. All I did was _Ctrl + Shift + P_ and in the Command Palette, typed Reload Window and hit _Enter_. My programming buddy was back ^_^

**Comment by vvvlad** [comment]
same issue for me on MacOS:
copilot 1.7.3.733
vscode 1.62.0
Output is empty. 
Tried to re-install, re-authenticate - nothing helps.
On my windows system it started working on first try without any issues.

Are there any other logs that can indicate the source of the issue?

**Reply by gj1118** [reply]
@vvvlad  were you able to resolve this issue please ? I am having the same issue as yours .. Please confirm as and when you can.. thanks

**Comment by gj1118** [comment]
Under accounts ( in the Activity sidebar ) if I choose my github account and choose "Manage Trusted Extensions" , I can see the Github Copilot is not using this account . Is this what is causing this issue ?? Yet this is the same account I had registered with initially when I had registered myself with Copilot.

![image](https://user-images.githubusercontent.com/473101/143659709-29e0dda5-6a9f-4b6e-a4d6-f94589894d2c.png)

Thanks
Gagan

**Reply by hansen0086** [reply]
I was having the same issue when trying to use the example on the copilot page (calculate days between dates in JavsScript) but it didn't offer any choices. But it works perfectly when I tried bubble sort in python.

**Comment by hoolymama** [comment]
For me, the issue seemed to be solved when I clicked the notification icon and agreed to the telemetry ToS
<img width="500" alt="Screen Shot 2022-04-30 at 12 57 57 PM" src="https://user-images.githubusercontent.com/83855/166117102-27ea4a3f-51cd-4780-967d-1183852ea320.png">

**Reply by Calderdavid** [reply]
thanks bro!!
